### DISCUSSION TITLE
I Can not run test on TestNG Platform="android" mvn clean -Dtest=Runner test

### DISCUSSION THREAD

**Discussion by chikhanh123** [AppiumTestDistribution/AppiumTestDistribution#1165] [General]
env: I use appium v2.4.0
[Appium] Available drivers:
[Appium]   - uiautomator2@2.37.1 (automationName 'UiAutomator2')
[Appium]   - xcuitest@5.12.2 (automationName 'XCUITest')
[Appium] Available plugins:
[Appium]   - device-farm@8.4.6 (ACTIVE)
[Appium]   - appium-dashboard@v2.0.2 (ACTIVE)

I download code https://github.com/AppiumTestDistribution/PageObjectPatternAppium
after that run :Platform="android" mvn clean -Dtest=Runner test
system return: 
![image](https://github.com/AppiumTestDistribution/AppiumTestDistribution/assets/43662553/e5e400fd-a6e8-42f7-9997-45b23c2c0995)
can anyone help me resolve bug?

**Comment by saikrishna321** [comment]
Please update to latest ATD version n try

**Reply by chikhanh123** [reply]
@saikrishna321. I had updated ATD 14.0.3 and then run: Platform="android" mvn clean -Dtest=Runner test
I have new issue: unable to call device farm endpoints java.io.IOException: HTTP/1.1 header parser received no bytes
![image](https://github.com/AppiumTestDistribution/AppiumTestDistribution/assets/43662553/4cdad604-8ad1-47d5-84b6-b3b4b1bf1777)

**Reply by saikrishna321** [reply]
@chikhanh123 please share full logs

**Reply by chikhanh123** [reply]
@saikrishna321 This is full log: 
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running com.appium.tests.Runner
SLF4J: No SLF4J providers were found.
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See https://www.slf4j.org/codes.html#noProviders for further details.
[Appium] Attempting to load plugin device-farm...
[Appium] Requiring plugin at /Users/mfv-computer-0081/node_modules/appium-device-farm/lib/src/index.js
[Appium] DevicePlugin has been successfully loaded in 2.686s
[Appium] Welcome to Appium v2.4.0
[Appium] Non-default server args:
[Appium] { address: '127.0.0.1',
[Appium]   basePath: '/wd/hub/',
[Appium]   port: 11460,
[Appium]   relaxedSecurityEnabled: true,
[Appium]   usePlugins: [ 'device-farm' ] }
[Appium] The autodetected Appium home path: /Users/mfv-computer-0081
[Appium] Attempting to load driver uiautomator2...
[Appium] Attempting to load driver xcuitest...
[Appium] Requiring driver at /Users/mfv-computer-0081/node_modules/appium-uiautomator2-driver/build/index.js
[Appium] AndroidUiautomator2Driver has been successfully loaded in 2.019s
[Appium] Requiring driver at /Users/mfv-computer-0081/node_modules/appium-xcuitest-driver/build/index.js
[Appium] XCUITestDriver has been successfully loaded in 4.317s
[device-farm-main] 📱 Update server with CLI Args: {"subcommand":"server","address":"127.0.0.1","basePath":"/wd/hub/","logFile":"/Users/mfv-computer-0081/Downloads/PageObjectPatternAppium-master/target/appiumlogs/appium_logs.txt","port":11460,"relaxedSecurityEnabled":true,"configFile":"/Users/mfv-computer-0081/Downloads/PageObjectPatternAppium-master/target/server.json","extraArgs":[],"usePlugins":["device-farm"],"plugin":{"device-farm":{"skipChromeDownload":true,"platform":"android","androidDeviceType":"both","iosDeviceType":"both"}},"allowCors":false,"allowInsecure":[],"callbackPort":4723,"debugLogSpacing":false,"denyInsecure":[],"keepAliveTimeout":600,"localTimezone":false,"loglevel":"debug","logNoColors":false,"logTimestamp":false,"pluginsImportChunkSize":7,"driversImportChunkSize":3,"longStacktrace":false,"noPermsCheck":false,"sessionOverride":false,"strictCaps":false,"useDrivers":[],"driver":{"xcuitest":{"wdaLocalPort":8100}},"tmpDir":"/var/folders/lp/7zwm2tm938b6pvvhzyklhqnr0000gq/T"}
[device-farm-main] 📱 Update server with Plugin Args: {"platform":"android","iosDeviceType":"both","androidDeviceType":"both","skipChromeDownload":true,"maxSessions":8,"adbRemote":[],"emulators":[],"simulators":[],"deviceAvailabilityTimeoutMs":300000,"deviceAvailabilityQueryIntervalMs":10000,"sendNodeDevicesToHubIntervalMs":30000,"checkStaleDevicesIntervalMs":30000,"checkBlockedDevicesIntervalMs":30000,"newCommandTimeoutSec":60,"bindHostOrIp":"192.168.1.15"}
[device-farm-main] proxy is not required for axios
[device-farm-main] Registering proxy middleware
[device-farm-main] Initializing database
[device-farm-main] Creating new database: ./temp-appium/db.json
[device-farm-main] Database loaded
[device-farm-main] Initializing storage
[device-farm-main] 📣📣📣 Device Farm Plugin will be served at 🔗 http://192.168.1.15:11460/device-farm
[device-farm-main] 📣📣📣 I'm a hub and I'm listening on 192.168.1.15:11460
[device-farm-main] Found 0 device candidates to be released
[device-farm-main] Hub will clean pending sessions every 30000 ms with pending session timeout: 310000 ms
[ADB] Found 1 'build-tools' folders under '/Users/mfv-computer-0081/Library/Android/sdk' (newest first):
[ADB]     /Users/mfv-computer-0081/Library/Android/sdk/build-tools/34.0.0
[ADB] Using 'adb' from '/Users/mfv-computer-0081/Library/Android/sdk/platform-tools/adb'
[ADB] Running '/Users/mfv-computer-0081/Library/Android/sdk/platform-tools/adb -P 5037 start-server'
[ADB] Getting connected devices
[ADB] Connected devices: [{"udid":"emulator-5554","state":"device"}]
[device-farm-main] fetchAndroidDevices from host: undefined. Found 1 android devices
[device-farm-main] Android Device details for emulator-5554 not available. So querying now.
[device-farm-main] skipChromeDownload server arg is set; skipping Chromedriver installation.
[device-farm-main] Android web/hybrid testing will not be possible without Chromedriver.
[ADB] Running '/Users/mfv-computer-0081/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell getprop ro.build.version.release'
[ADB] Running '/Users/mfv-computer-0081/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell getprop ro.build.characteristics'
[ADB] Running '/Users/mfv-computer-0081/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell getprop ro.product.name'
[device-farm-main] Found 1 android devices
[device-farm-main] Added 1 new devices to local database
[device-farm-main] Unblocked device emulator-5554 from host http://192.168.1.15:11460
[Appium] Appium REST http interface listener started on http://127.0.0.1:11460/wd/hub
[Appium] Available drivers:
[Appium]   - uiautomator2@2.37.1 (automationName 'UiAutomator2')
[Appium]   - xcuitest@5.12.2 (automationName 'XCUITest')
[Appium] Available plugins:
[Appium]   - device-farm@8.4.6 (ACTIVE)
[Appium]   - appium-dashboard@v2.0.2
[HTTP] --> GET /wd/hub/status
[HTTP] {}
[AppiumDriver@0ec9] Calling AppiumDriver.getStatus() with args: []
[device-farm-main] 📱 Plugin Args: {"skipChromeDownload":true,"platform":"android","androidDeviceType":"both","iosDeviceType":"both"}
[AppiumDriver@0ec9] Plugins which can handle cmd 'getStatus': device-farm
[AppiumDriver@0ec9] Plugin device-farm is now handling cmd 'getStatus'
[AppiumDriver@0ec9] Executing default handling behavior for command 'getStatus'
[AppiumDriver@0ec9] Responding to client with driver.getStatus() result: {"ready":true,"message":"The server is ready to accept new connections","build":{"version":"2.4.0"}}
[HTTP] <-- GET /wd/hub/status 200 9 ms - 110
[HTTP] 
1
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 13.87 s <<< FAILURE! -- in com.appium.tests.Runner
[ERROR] com.appium.tests.Runner.testRunner -- Time elapsed: 13.31 s <<< FAILURE!
java.lang.RuntimeException: unable to call device farm endpoints java.io.IOException: HTTP/1.1 header parser received no bytes
        at com.appium.utils.Api.getResponse(Api.java:29)
        at com.appium.device.Devices.getConnectedDevices(Devices.java:26)
        at com.appium.manager.ATDRunner.<init>(ATDRunner.java:41)
        at com.appium.tests.Runner.testRunner(Runner.java:13)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
        at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:677)
        at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:221)
        at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
        at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:969)
        at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:194)
        at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
        at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
        at org.testng.TestRunner.privateRun(TestRunner.java:829)
        at org.testng.TestRunner.run(TestRunner.java:602)
        at org.testng.SuiteRunner.runTest(SuiteRunner.java:437)
        at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:431)
        at org.testng.SuiteRunner.privateRun(SuiteRunner.java:391)
        at org.testng.SuiteRunner.run(SuiteRunner.java:330)
        at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
        at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
        at org.testng.TestNG.runSuitesSequentially(TestNG.java:1256)
        at org.testng.TestNG.runSuitesLocally(TestNG.java:1176)
        at org.testng.TestNG.runSuites(TestNG.java:1099)
        at org.testng.TestNG.run(TestNG.java:1067)
        at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:155)
        at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.executeSingleClass(TestNGDirectoryTestSuite.java:102)
        at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.execute(TestNGDirectoryTestSuite.java:91)
        at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:137)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
        at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
        at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   Runner.testRunner:13 » Runtime unable to call device farm endpoints java.io.IOException: HTTP/1.1 header parser received no bytes
[INFO] 
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  26.340 s
[INFO] Finished at: 2024-01-05T14:05:11+07:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.1.2:test (default-test) on project PagePatternAppium: There are test failures.
[ERROR] 
[ERROR] Please refer to /Users/mfv-computer-0081/Downloads/PageObjectPatternAppium-master/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
mfv-computer-0081@MacBook-Pro-Kieran-9 PageObjectPatternAppium-master %
